lib/cog/bus_driver.ex

```diff
@@ -47,34 +49,38 @@ defmodule Cog.BusDriver do
-    mqtt_port = Keyword.get(bus_opts, :port, 1883)
-    cert_file = Keyword.get(bus_opts, :ssl_cert)
-    key_file = Keyword.get(bus_opts, :ssl_key)
-    bindings = [mqtt_addr: mqtt_host, mqtt_port: mqtt_port]
-    case load_private_config("common_mqtt") do
-      {:ok, _} ->
-        cond do
-          # SSL enabled
-          cert_file != nil and key_file != nil ->
-            Logger.info("Message bus configured for SSL")
-            load_private_config("ssl_mqtt", [cert: String.to_char_list(cert_file),
-                                             key: String.to_char_list(key_file),
-                                             mqtt_type: :mqtts] ++ bindings)
-          # SSL disabled
-          cert_file == nil and key_file == nil ->
-            Logger.info("Message bus configured for plain TCP")
-            load_private_config("plain_mqtt", [{:mqtt_type, :mqtt}|bindings])
-          # SSL misconfigured (either cert_file is nil or key_file is nil)
-          true ->
-            if cert_file == nil do
-              Logger.error("Message bus SSL configuration error. Path to certificate file is empty.")
-              {:error, {:missing_config, :cert_file}}
-            else
-              Logger.error("Message bus SSL configuration error. Path to key file is empty.")
-              {:error, {:missing_config, :key_file}}
-            end
-      end
+    case prepare_host(Keyword.get(bus_opts, :host, "127.0.0.1")) do
+      {:ok, mqtt_host} ->
+        mqtt_port = Keyword.get(bus_opts, :port, 1883)
+        cert_file = Keyword.get(bus_opts, :ssl_cert)
+        key_file = Keyword.get(bus_opts, :ssl_key)
+        bindings = [mqtt_addr: mqtt_host, mqtt_port: mqtt_port]
+        case load_private_config("common_mqtt") do
+          {:ok, _} ->
+            cond do
+              # SSL enabled
+              cert_file != nil and key_file != nil ->
+                Logger.info("Message bus configured for SSL")
+                load_private_config("ssl_mqtt", [cert: String.to_char_list(cert_file),
+                                                 key: String.to_char_list(key_file),
+                                                 mqtt_type: :mqtts] ++ bindings)
+                # SSL disabled
+                cert_file == nil and key_file == nil ->
+                Logger.info("Message bus configured for plain TCP")
+                load_private_config("plain_mqtt", [{:mqtt_type, :mqtt}|bindings])
+                # SSL misconfigured (either cert_file is nil or key_file is nil)
+                true ->
+                if cert_file == nil do
+                  Logger.error("Message bus SSL configuration error. Path to certificate file is empty.")
+                  {:error, {:missing_config, :cert_file}}
+                else
+                  Logger.error("Message bus SSL configuration error. Path to key file is empty.")
+                  {:error, {:missing_config, :key_file}}
+                end
+          end
```

Comment:
Looks like a little bit of weird indentation above. Might be nice to combine the two conditionals above into a single `case` but not a big deal.